storage/gcloud/aio/storage/storage.py

```diff
@@ -490,11 +490,15 @@ async def _upload_multipart(self, url: str, object_name: str,
                                 timeout: int = 30) -> Dict[str, Any]:
         # https://cloud.google.com/storage/docs/json_api/v1/how-tos/multipart-upload
         params['uploadType'] = 'multipart'
-
         metadata_headers = {'Content-Type': 'application/json; charset=UTF-8'}
-        metadata = {self._format_metadata_key(k): v
-                    for k, v in metadata.items()}
-        metadata['name'] = object_name
+        metadict = (metadata or {}).copy()
+        custom_metadata = metadict.pop('metadata', None)
+        metadata_content = {self._format_metadata_key(k): v
+                            for k, v in metadict.items()}
+
+        metadata_content['name'] = object_name
+        if custom_metadata:
+            metadata_content['metadata'] = custom_metadata
```

Comment:
Could you explain what this code change is meant to accomplish? It seems you're pulling `metadata` out of the dict to avoid running it through `_format_metadata_key()` and re-adding it later, but `_format_metadata_key('metadata') == 'metadata'` so I'm a bit confused as to why you need to do this?

Reply:
First off all, thanks for this library.

And you're right, this is absolutely not necessary. This is an artefact of my experimentations, and I will remove it.